lib/marin/src/marin/execution/executor.py

```diff
@@ -1131,6 +1147,31 @@ def _setup_logging():
     logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
 
 
+def _detect_local_resources() -> dict[str, float]:
+    from marin.resources_utils import torch_device_name_to_ray_accel_type
+
+    resources: dict[str, float] = {"head_node": 1.0}
+    try:
+        import torch
+
+        if not torch.cuda.is_available():
+            return resources
+        gpu_count = torch.cuda.device_count()
+        if gpu_count == 0:
+            return resources
+        # TODO: assuming same GPU type
+        device_name = torch.cuda.get_device_name(0)
+        accel_type = torch_device_name_to_ray_accel_type(device_name)
+        if accel_type:
+            logger.info(f"Auto-detected {gpu_count} GPU(s): '{device_name}' -> accelerator_type:{accel_type}")
+            resources[f"accelerator_type:{accel_type}"] = gpu_count
+        else:
+            logger.warning(f"Could not map GPU '{device_name}' to a known Ray accelerator type.")
```

Comment:
i'd rather we didn't sniff the gpu type here and instaed work with num_gpus=1 by default

Reply:
resolved in Discord conversations (https://discord.com/channels/1354881461060243556/1445592865009827871, https://discord.com/channels/1354881461060243556/1444945051259244606). Need for Ray local mode scheduling.